src/qililab/platform/platform.py

```diff
@@ -652,6 +652,11 @@ def _execute_qprogram_with_qblox(
                 buses[bus_alias].upload_qpysequence(qpysequence=sequences[bus_alias])
                 self._qpy_sequence_cache[bus_alias] = sequence_hash
 
+        # sync all instruments
+        for instrument in self.instruments.elements:
+            if isinstance(instrument, QbloxModule):
+                instrument.sync_sequencers()
```

Comment:
If you do this, you will enable syncing to **all** sequencers present. We only want to enable syncing to the sequencers that are relevant to the specific QProgram.

The de-sync is happening for all sequencers, since you don't really care about it (it doesn't affect anyone if you desync an irrelevant sequencer), but you may change that too so it's more consistent.

Reply:
Now I set the sync as true inside the bus loop (`for bus_alias in sequences:`) where the `upload_qpysequence` is located (it is not inside the conditional) with:
```
for instrument in buses[bus_alias].system_control.instruments:
    if isinstance(instrument, QbloxModule):
        instrument.sync_sequencers(buses[bus_alias].port)
```
All the required information (port and instruments) comes from the bus

And now the desync is also like this:
```
# Reset instrument settings
for bus_alias in sequences:
    for instrument in buses[bus_alias].system_control.instruments:
        if isinstance(instrument, QbloxModule):
            instrument.desync_sequencers(buses[bus_alias].port)
```

The functions changed inside QbloxModule (since these are only rellevant here) are:
```
def sync_sequencers(self, port: str) -> None:
    """Syncs all sequencers."""
    sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
    for sequencer in sequencers:
        self.device.sequencers[sequencer.identifier].sync_en(True)

def desync_sequencers(self, port: str) -> None:
    """Syncs all sequencers."""
    sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=port)
    for sequencer in sequencers:
        self.device.sequencers[sequencer.identifier].sync_en(False)
```

Reply:
I have also tested the syncing

![imagen](https://github.com/qilimanjaro-tech/qililab/assets/151619025/1b7c89d8-c00b-4c7d-b20e-65a777dffbee)

but the code saving still gives me errors by changing the imports